services/blockstorage/service.go

```diff
@@ -82,34 +102,27 @@ func (s *service) loadTransactionsBlockHeader(height primitives.BlockHeight) (*s
 }
 
 func (s *service) GetTransactionsBlockHeader(input *services.GetTransactionsBlockHeaderInput) (*services.GetTransactionsBlockHeaderOutput, error) {
-	if input.BlockHeight > s.lastCommittedBlockHeight && input.BlockHeight-s.lastCommittedBlockHeight <= 5 {
-		c := make(chan *services.GetTransactionsBlockHeaderOutput)
-
-		go func() {
-			const interval = 10 * NanosecondsInMillisecond
-			timeout := s.config.BlockSyncCommitTimeout().Nanoseconds()
-
-			for i := int64(0); i < timeout; i += interval {
+	currentBlockHeight := s.lastCommittedBlockHeight
+	if input.BlockHeight > currentBlockHeight && input.BlockHeight-currentBlockHeight <= 5 {
+		timeout := time.NewTimer(s.config.BlockSyncCommitTimeout())
+		defer timeout.Stop()
+		tick := time.NewTicker(10 * time.Millisecond)
```

Comment:
Let's add a future TODO here to remove the polling altogether, we had several ideas of how it can be done.

My personal favorite is to create a channel for the result block height you wait for and then wait on it.

Then when we reach this block height, close the channel.

This way we'll always have a small map of 5 active channels for block heights that are still within our window.

Reply:
we have an open issue on this, i'll reference to it in the todo